test/functional/sql-tag/sql-tag-parameters-mysql.ts

```diff
@@ -0,0 +1,189 @@
+                    { id: "test1", value: 10, name: "a" },
+                    { id: "test2", value: 20, name: "b" },
+                    { id: "test3", value: 30, name: "c" },
+                ])
+
+                const minValue = 15
+                const maxValue = 25
+                const namePattern = "b"
+                const [example] = await connection.sql`SELECT * FROM example
+                    WHERE value > ${minValue}
+                    AND value < ${maxValue}
+                    AND name LIKE ${namePattern}
+                    ORDER BY value DESC`
+
+                expect(example.id).to.equal("test2")
+                expect(example.value).to.equal(20)
+            }),
+        ))
+
+    it("should handle SQL tag parameters with NULL values", () =>
+        Promise.all(
+            connections.map(async (connection) => {
+                const repo = connection.getRepository(MysqlExample)
+
+                await repo.save([
+                    { id: "null1", value: null },
+                    { id: "null2", value: 10 },
+                ])
+
+                const examples = await connection.sql`
+                    SELECT * FROM example WHERE value IS ${null}
+                `
+
+                const ids = examples.map((e: MysqlExample) => e.id)
+
+                expect(examples).to.have.length(1)
+                expect(ids).to.have.members(["null1"])
+            }),
+        ))
+
+    it("should handle SQL tag parameters with boolean values", () =>
+        Promise.all(
+            connections.map(async (connection) => {
+                const repo = connection.getRepository(MysqlExample)
+
+                await repo.save([
+                    { id: "true1", active: true },
+                    { id: "false1", active: false },
+                ])
+
+                const value = true
+
+                const examples = await connection.sql`
+                    SELECT * FROM example WHERE active = ${value}
+                `
+
+                const ids = examples.map((e: MysqlExample) => e.id)
+
+                expect(examples).to.have.length(1)
+                expect(ids).to.have.members(["true1"])
+            }),
+        ))
+
+    it("should handle SQL tag parameters with date values", () =>
+        Promise.all(
+            connections.map(async (connection) => {
+                const repo = connection.getRepository(MysqlExample)
+                const now = new Date()
+                const yesterday = new Date(now.getTime() - 24 * 60 * 60 * 1000)
+                const nowStr = toMySQLDateString(now)
+                const yesterdayStr = toMySQLDateString(yesterday)
+
+                await repo.save([
+                    { id: "today", createdAt: nowStr },
+                    { id: "yesterday", createdAt: yesterdayStr },
+                ])
+
+                const examples = await connection.sql`
+                    SELECT * FROM example WHERE createdAt > ${yesterdayStr}
```

Comment:
In the equivalent test for Postgres, the `Date` value `yesterday` is used directly as the parameter, whereas here it's converted to a string first. I would expect `Date` values to be handled the same across all drivers, so it would be good to show they work for MySQL here.

Reply:
I'm confused... why were Dates working in Postgres without any serialization before - I assumed we already had some serialization in place, but is it actually because Postgres supports the default `Date.toString()` format and other databases don't?

I don't think we should be changing how dates work in raw queries in this PR. Which means the only serialization required is the expansion of function parameters, which can be internal to `buildSqlTag`.

Reply:
Apparently it had to do with the fact that I had to explictly set the datetime precision. I removed that extra logic as I believe it doesn't belong in `.sql` which should simply be a thin wrapper around `.query`.